src/backend/utils/adt/ri_triggers.c

```diff
@@ -550,8 +549,8 @@ ri_Check_Pk_Match(Relation pk_rel, Relation fk_rel,
 		appendStringInfoString(&querybuf, " FOR KEY SHARE OF x");
 
 		/* Prepare and save the plan */
-		RUN_AS_PSQL(qplan = ri_PlanCheck(querybuf.data, riinfo->nkeys, queryoids,
```

Comment:
safest would be this

if (dialect == TSQL)
   RUN_AS_PSQL()
else
   current code

Reply:
We already have this check inside `RUN_AS_PSQL` macro. So we don't need to explicitly check this.